app/src/main/java/zapsolutions/zap/transactionHistory/transactionDetails/LnPaymentDetailBSDFragment.java

```diff
@@ -0,0 +1,170 @@
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.github.lightningnetwork.lnd.lnrpc.LightningGrpc;
+import com.github.lightningnetwork.lnd.lnrpc.PayReq;
+import com.github.lightningnetwork.lnd.lnrpc.PayReqString;
+import com.github.lightningnetwork.lnd.lnrpc.Payment;
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import zapsolutions.zap.R;
+import zapsolutions.zap.connection.establishConnectionToLnd.LndConnection;
+import zapsolutions.zap.util.ClipBoardUtil;
+import zapsolutions.zap.util.ExecuteOnCaller;
+import zapsolutions.zap.util.MonetaryUtil;
+import zapsolutions.zap.util.ZapLog;
+
+import java.text.DateFormat;
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+
+public class LnPaymentDetailBSDFragment extends BottomSheetDialogFragment {
+
+    public static final String TAG = LnPaymentDetailBSDFragment.class.getName();
+    public static final String ARGS_TRANSACTION = "TRANSACTION";
+
+    private TextView mTransactionDescription;
+    private TextView mAmountLabel;
+    private TextView mAmount;
+    private TextView mMemoLabel;
+    private TextView mMemo;
+    private TextView mFeeLabel;
+    private TextView mFee;
+    private TextView mDateLabel;
+    private TextView mDate;
+    private TextView mPreimageLabel;
+    private TextView mPreimage;
+    private ImageView mPreimageCopyIcon;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.bsd_payment_detail, container);
+
+        mTransactionDescription = view.findViewById(R.id.transactionDescription);
+        mAmountLabel = view.findViewById(R.id.amountLabel);
+        mAmount = view.findViewById(R.id.amount);
+        mMemoLabel = view.findViewById(R.id.memoLabel);
+        mMemo = view.findViewById(R.id.memo);
+        mFeeLabel = view.findViewById(R.id.feeLabel);
+        mFee = view.findViewById(R.id.fee);
+        mDateLabel = view.findViewById(R.id.dateLabel);
+        mDate = view.findViewById(R.id.date);
+        mPreimageLabel = view.findViewById(R.id.preimageLabel);
+        mPreimage = view.findViewById(R.id.preimage);
+        mPreimageCopyIcon = view.findViewById(R.id.preimageCopyIcon);
+
+        ImageButton closeButton = view.findViewById(R.id.closeButton);
+        closeButton.setOnClickListener(view1 -> dismiss());
+
+        if (getArguments() != null) {
+            ByteString transactionString = (ByteString) getArguments().getSerializable(ARGS_TRANSACTION);
+            try {
+                bindPayment(transactionString);
+            } catch (InvalidProtocolBufferException exception) {
```

Comment:
Could be simplified to `} catch (InvalidProtocolBufferException | NullPointerException exception) {`